src/crystal/system/win32/delay_load.cr

```diff
@@ -0,0 +1,205 @@
+require "c/delayimp"
+
+private ERROR_SEVERITY_ERROR = 0xC0000000_u32
+private FACILITY_VISUALCPP   =           0x6d
+
+private macro vcpp_exception(err)
+  {{ ERROR_SEVERITY_ERROR | (FACILITY_VISUALCPP << 16) | err }}
+end
+
+lib LibC
+  $image_base = __ImageBase : IMAGE_DOS_HEADER
+end
+
+private macro p_from_rva(rva)
```

Comment:
There is just one caveat: for some reason `LibC.image_base` cannot be accessed from any LLVM module scope other than the top-level one, otherwise MSVC raises an internal linker error. So this cannot be a class method of `Crystal::System::DelayLoad`, and neither can `__delayLoadHelper2` itself